scripts/validate_docstrings.py

```diff
@@ -331,6 +335,26 @@ def parameter_mismatches(self):
 
         return errs
 
+    def validate_pep8(self):
+        create_function = 'def {name}():\n' \
+                          '    """\n{examples}\n    """\n' \
+                          '    pass\n'
+
+        name = self.name.split('.')[-1]
+        lines = (' ' * 4 + line if line else '' for line in self.examples)
+        examples = '\n'.join(lines)
+        with tempfile.NamedTemporaryFile(mode='w') as file:
+            func = create_function.format(name=name, examples=examples)
+            file.write(func)
+            file.flush()
+
+            application = flake8.Application()
+            application.initialize(["--doctests", "--quiet"])
+            application.run_checks([file.name])
```

Comment:
can you just leave the writing to the file and this line inside the context manager, and creating the app object, initializing... have it outside?